src/Avalonia.Animation/Animation.cs

```diff
@@ -153,5 +160,31 @@ public IDisposable Apply(Animatable control, IObservable<bool> matchObs)
             }
             return this;
         }
+
+        /// <inheritdocs/>
+        public Task RunAsync(Animatable control)
+        {
+            var run = new TaskCompletionSource<object>();
+
+            if (this.RepeatCount == RepeatCount.Loop)
+                run.SetException(new InvalidOperationException("Looping animations must not use the Run method."));
+
+            this.Done += (sender, args) =>
+            {
+                if (sender == control)
+                    run.SetResult(null);
+                else
+                    run.SetCanceled();
```

Comment:
Why are we canceling the task here? This wouldn't allow users to run the same animation on two different controls at the same time.

Reply:
I was thinking of making it cancel the task when an incongruity occurs but yeah, it's the wrong behaviour